agent/agent.go

```diff
@@ -503,6 +503,8 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string)
 		err := a.uploadToS3(ctx, tc)
 		tc.logger.Execution().Error(errors.Wrap(err, "error uploading log files"))
 		tc.logger.Execution().Infof("Sending final status as: %v", detail.Status)
+		ctx, cancel = a.withCallbackTimeout(ctx, tc)
```

Comment:
This seems like kind of a long while to wait for flogs to flush. I think this defaults to 15 minutes.

Reply:
I think it's 15 minutes, or a project-defined timeout. I can change it to 1 minute?